applet/src/main.c

```diff
@@ -144,17 +145,67 @@ void demo(){
 
   drawtext(VERSIONDATE,
            160,220);
-  sleep(1000);
-  
-  //Make the welcome image scroll across the screen.
+  if (quick) {
+    sleep(500);
+  } else {
+    sleep(1000);
+  }
+}
+
+//Make the welcome image scroll across the screen.
+void demo_show_animation(void) {
   for(int i=0;i<0x60;i+=3){
     gfx_drawbmp(welcomebmp,0,i);
     sleep(30);
   }
-#endif //CONFIG_GRAPHICS
+}
+
+void demo_clear(void) {
+  // Clear screen
+  unsigned int oldfg = gfx_info.fg_color;
```

Comment:
gfx_info is undefined under D02 firmware. so you should exclude reference to it with ifdefs.